src/main/java/com/johan/blignaut/entelect/vitalityincubator/task2/GeoDBController.java

```diff
@@ -0,0 +1,42 @@
+package com.johan.blignaut.entelect.vitalityincubator.task2;
+
+import com.johan.blignaut.entelect.vitalityincubator.restclient.api.GeodbApi;
+import com.johan.blignaut.entelect.vitalityincubator.restclient.model.GeoDBClient;
+import com.johan.blignaut.entelect.vitalityincubator.restserver.api.GeoApi;
+import com.johan.blignaut.entelect.vitalityincubator.restserver.model.GeoDBServer;
+import com.johan.blignaut.entelect.vitalityincubator.task2.exception.CustomInternalServerException;
+import com.johan.blignaut.entelect.vitalityincubator.task2.service.GeoDBMappingService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+import retrofit2.Call;
+import retrofit2.Response;
+
+import java.io.IOException;
+
+@RestController
+public class GeoDBController implements GeoApi {
+
+    private final GeodbApi geodbApi;
+    private final GeoDBMappingService geoDBMappingService;
+
+    public GeoDBController(GeodbApi geodbApi, GeoDBMappingService geoDBMappingService) {
+        this.geodbApi = geodbApi;
+        this.geoDBMappingService = geoDBMappingService;
+    }
+
+    @Override
+    public ResponseEntity<GeoDBServer> getPlaces(String filterByName) {
+        try {
+            Call<GeoDBClient> call = geodbApi.getPlaces();
```

Comment:
The logic for the service call and error checking should stay in the wrapper class (GeoDBMappingHelper).
So you'd call the method in GeoDBMappingHelper and get the `geodbClient` response, and then pass that to the mapper class like you've done here.

Reply:
I don't understand, what should be in this `getPlaces` method then?

Reply:
`getPlaces` in the controller should be something like:

```
geoDBServerResponse = geoDBMappingHelper.getGeoDBServerResponse()
return ResponseEntity.ok(geoDBMappingService.mapClientToServerGeoDB(geoDBServerResponse))

```

The method in the controller class needs very little logic.
The service wrapper class (geoDBMappingHelper) handles the downstream service call and validating the response.
The mapper class does the mapping.

I'll approve the PR but please address this on your next PR.

Reply:
Oh okay thanks. I have already addressed it, you can view it already, this comment is on an outdated version of the file.